msal/src/Microsoft.Identity.Client/Platforms/net45/PlatformInformation.cs

```diff
@@ -57,6 +57,28 @@ public override string GetProductName()
             return "MSAL.Desktop";
         }
 
+        public override async Task<string> GetUserPrincipalNameAsync()
+        {
+            return await Task.Factory.StartNew(() =>
+            {
+                const int NameUserPrincipal = 8;
+                uint userNameSize = 0;
+                NativeMethods.GetUserNameEx(NameUserPrincipal, null, ref userNameSize);
```

Comment:
>NativeMethods.GetUserNameEx(NameUserPrincipal, null, ref userNameSize); [](start = 16, length = 71)

I'm surprised that .net doesn't support getting this

Reply:
Perhaps because this is platform-specific? I copied same implementation from ADAL, which was last reviewed in [this different PR 6 days ago](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1153/files#diff-7b0de510ef9cb6269976fee57e0eaa1fR53).

---
In reply to: [207390938](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r207390938) [](ancestors = 207390938)